index.md

```diff
@@ -30,30 +30,33 @@ and accounting of disclosures, are outside the scope of this profile.
 
 ## Support for "public" and "confidential" apps
 
-Within this profile we differentiate between two types of 
-apps based upon whether the execution environment within which the app runs 
-enables the app to protect secrets.   Pure client-side apps 
-(for example, HTML5/JS browser-based apps, iOS mobile
-apps, or Windows desktop apps) can provide adequate security -- but they can't
-"keep a secret" in the OAuth2 sense. That is to say, any "secret" key, code, or
-string that's embedded in the app can potentially be extracted by an end-user
-or attacker. So security for these apps can't depend on secrets embedded at
-install-time. Security assurance comes from being hosted within a trusted 
-server environment.
+Within this profile we differentiate between two [types of apps](https://tools.ietf.org/html/rfc6749#section-2.1)
+based upon whether the execution environment within which the app runs enables
+the app to protect secrets. Pure client-side apps (for example, HTML5/JS
+browser-based apps, iOS mobile apps, or Windows desktop apps) can provide
+adequate security, but they may be unable to “keep a secret” in the OAuth2
+sense, that is to say, any “secret” key, code, or string that's statically
+embedded in the app can potentially be extracted by an end-user or attacker.
+Hence security for these apps can't depend on secrets embedded at install-time.
 
-#### Use the <span class="label label-primary">confidential app</span>  profile
-when all of the following apply:
+For strategies and best practices to protecting a client secret refer to:
 
-* App runs on a trusted server
-* App has server-side business logic (e.g. using PHP, Python, Ruby, .NET, etc.)
-* App is *able to protect* a `client_secret`
+- OAuth 2.0 Threat Model and Security Considerations: [4.1.1. Threat: Obtaining Client Secrets](https://tools.ietf.org/html/rfc6819#section-4.1.1)
+- OAuth 2.0 for Native Apps: [8.5. Client Authentication](https://tools.ietf.org/html/draft-ietf-oauth-native-apps-12#section-8.5)
+- [OAuth 2.0 Dynamic Client Registration Protocol](https://tools.ietf.org/html/rfc7591)
 
 
+#### Use the <span class="label label-primary">confidential app</span> profile
+if your app is *able to protect* a `client_secret`, e.g.:
+
+* App runs on a trusted server with only server-side access to the secret
+* App is a native app that uses more complex technologies (such as dynamic client registration and universal redirect_uris) to protect the `client_secret`
+
 #### Use the <span class="label label-primary">public app</span> profile 
-when all of the following apply:
+if your app is *unable to protect* a `client_secret`, e.g.:
 
-* App runs on an end-user's device (e.g. HTML5/JS in-browser; native iOS, Windows, or Android)
-* App is *unable to protect* a `client_secret`
+* App is an HTML5/JS in-browser app that would expose the secret in user space
```

Comment:
App is an HTML5 **or** JS in-browser app that would expose the secret in user space